RecoBTag/SecondaryVertex/src/CandidateBoostedDoubleSecondaryVertexComputer.cc

```diff
@@ -54,111 +95,474 @@ float CandidateBoostedDoubleSecondaryVertexComputer::discriminator(const TagInfo
-    math::XYZTLorentzVector vertexSum = vtxKinematics.weightedVectorSum();
     if (cont==1)
     {
-      SV_mass_0 = vertex.p4().mass()  ;
-      SV_EnergyRatio_0 = vertexSum.E() / allSum.E();
-      SV_pt_0 = vertex.p4().pt();
       flightDir_0 = svTagInfo.flightDirection(iVtx->second);
       SV_p4_0 = vertex.p4();
+      vtxMass = SV_p4_0.mass();
 
-      if (reco::deltaR2(flightDir_0,currentAxes[1])<reco::deltaR2(flightDir_0,currentAxes[0]))
-        tau_dot = (currentAxes[1].px()*flightDir_0.x()+currentAxes[1].py()*flightDir_0.y()+currentAxes[1].pz()*flightDir_0.z())/(sqrt(currentAxes[1].modp2())*flightDir_0.mag());
-      else
-        tau_dot = (currentAxes[0].px()*flightDir_0.x()+currentAxes[0].py()*flightDir_0.y()+currentAxes[0].pz()*flightDir_0.z())/(sqrt(currentAxes[0].modp2())*flightDir_0.mag());
+      if(vtxMass > 0.)
+        z_ratio = reco::deltaR(currentAxes[1],currentAxes[0])*SV_p4_0.pt()/vtxMass;
     }
     if (cont==2)
     {
-      SV_EnergyRatio_1= vertexSum.E() / allSum.E();
       flightDir_1 = svTagInfo.flightDirection(iVtx->second);
       SV_p4_1 = vertex.p4();
-      z_ratio = reco::deltaR(flightDir_0,flightDir_1)*SV_pt_0/(SV_p4_0+SV_p4_1).mass();
+      vtxMass = (SV_p4_1+SV_p4_0).mass();
+
+      if(vtxMass > 0.)
+        z_ratio = reco::deltaR(flightDir_0,flightDir_1)*SV_p4_1.pt()/vtxMass;
+
       break;
     }
   }
 
-  nSM = muonTagInfo.leptons();
-  nSE = elecTagInfo.leptons();
-  nSL = nSM + nSE;
+  // when only one tau axis has SVs assigned, they are all assigned to the 1st tau axis
+  // in the special case below need to swap values
+  if( (tau1_vertexMass<0 && tau2_vertexMass>0) )
+  {
+    float temp = tau1_trackEtaRel_0;
+    tau1_trackEtaRel_0= tau2_trackEtaRel_0;
+    tau2_trackEtaRel_0= temp;
 
-  float PFLepton_ptrel = -1., PFLepton_IP2D = -1.;
+    temp = tau1_trackEtaRel_1;
+    tau1_trackEtaRel_1= tau2_trackEtaRel_1;
+    tau2_trackEtaRel_1= temp;
 
-  // PFMuon information
-  for (size_t leptIdx = 0; leptIdx < muonTagInfo.leptons() ; ++leptIdx)
-  {
-    float PFMuon_ptrel = (muonTagInfo.properties(leptIdx).ptRel);
-    if (PFMuon_ptrel > PFLepton_ptrel )
-    {
-      PFLepton_ptrel = PFMuon_ptrel;
-      PFLepton_IP2D  = (muonTagInfo.properties(leptIdx).sip2d);
-    }
-  }
+    temp = tau1_trackEtaRel_2;
+    tau1_trackEtaRel_2= tau2_trackEtaRel_2;
+    tau2_trackEtaRel_2= temp;
 
-  // PFElectron information
-  for (size_t leptIdx = 0; leptIdx <  elecTagInfo.leptons() ; ++leptIdx)
-  {
-    float PFElectron_ptrel = (elecTagInfo.properties(leptIdx).ptRel);
-    if (PFElectron_ptrel > PFLepton_ptrel )
-    {
-      PFLepton_ptrel = PFElectron_ptrel;
-      PFLepton_IP2D  = (elecTagInfo.properties(leptIdx).sip2d);
-    }
+    temp = tau1_flightDistance2dSig;
+    tau1_flightDistance2dSig= tau2_flightDistance2dSig;
+    tau2_flightDistance2dSig= temp;
+
+    temp = tau1_vertexDeltaR;
+    tau1_vertexDeltaR= tau2_vertexDeltaR;
```

Comment:
`tau2_vertexDeltaR` is not used later so `tau2_vertexDeltaR = temp` would be a dead assignment and was removed in one of the previous commit after you made a comment about. You are now making a valid point that `temp = tau1_vertexDeltaR` is pointless and should be deleted. This is fixed now.